src/score_hv/harvesters/inc_logs.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python
+
+""" Collection of methods to faciliate retrieval of increment
+    descriptive statistics from logs
+"""
+
+import os
+
+from collections import namedtuple
+from dataclasses import dataclass
+from dataclasses import field
+
+from score_hv.config_base import ConfigInterface
+
+HRVSTER_NAME = 'inc_logs'
+VALID_STATISTICS = ('mean', 'RMS')
+VALID_VARIABLES = ['pt_inc', 's_inc', 'u_inc', 'v_inc', 'SSH', 'Salinity',
+                   'Temperature', 'Speed of Currents', 'o3mr_inc', 'sphum_inc',
+                   'T_inc', 'delp_inc', 'delz_inc']
+
+HarvestedData = namedtuple('HarvestedData', ['logfile',
+                                             'statistic',
+                                             'variable',
+                                             'value', 
+                                             'units'])
+N_TUPLE_ENTRIES = 5                                             
+
+@dataclass
+class LogIncCfg(ConfigInterface):
+    """ Dataclass to hold and provide configuration information pertaining to
+        how the harvester should read increment log files
+    
+        Parameters:
+        -----------
+        config_data: dict - contains configuration data parsed from either an
+                            input yaml file or input dict
+        
+    """
+    config_data: dict = field(default_factory=dict)
+    
+    def __post_init__(self):
+        self.set_config()
+    
+    def set_config(self):
+        """ function to set configuration variables from given dictionary
+        """ 
+        try:
+            self.harvest_filename = self.config_data.get('filename')
+        except Exception as err:
+            msg = '%s key missing' % key
```

Comment:
where is the key value set for this message? it doesn't make sense now so I would guess if this error came up it would be unclear in regards to the code

Reply:
nice catch - I'm not actually sure what the purpose of this try / except block is

Removing the try except block here to allow the standard errors to output more helpful messages in this case